### PR TITLE
✨  Ads RTC: support amp-script fetching

### PR DESCRIPTION
**Pull request by samouri**
**summary** 
Fixes https://github.com/ampproject/amphtml/issues/33581.

- Adds support for RTC for fetching JSON from `amp-script`. 
- Unit test + manual example both included
- ~Blocked on https://github.com/ampproject/amphtml/pull/33900~

### CODE REVIEW

File: src/service/real-time-config/real-time-config-impl.js [hunk 1]
```diff
@@ -589,8 +624,8 @@ export class RealTimeConfigManager {
     this.modifyRtcConfigForConsentStateSettings();
     customMacros = this.assignMacros(customMacros);
     this.rtcStartTime_ = Date.now();
-    this.handleRtcForCustomUrls(customMacros, checkStillCurrent);
-    this.handleRtcForVendorUrls(customMacros, checkStillCurrent);
+    this.handleRtcForCustomUrls(customMacros, checkStillCurrent, element);
+    this.handleRtcForVendorUrls(customMacros, checkStillCurrent, element);
```
Comment (samouri) [inline src/service/real-time-config/real-time-config-impl.js:628]: nit: can we assume amp-script will never be used by vendor urls?

File: src/service/real-time-config/real-time-config-impl.js [hunk 2]
```diff
@@ -502,45 +515,67 @@ export class RealTimeConfigManager {
...
+    if (isAmpScriptUri(url)) {
+      rtcFetch = Services.scriptForDocOrNull(element)
+        .then((service) => {
+          userAssert(service, 'AMP-SCRIPT is not installed.');
+          return service.fetch(url);
+        })
+        .then((json) => {
+          checkStillCurrent();
+          const rtcTime = Date.now() - this.rtcStartTime_;
+          if (typeof json !== 'object') {
```
Comment (calebcordry) [inline src/service/real-time-config/real-time-config-impl.js:531]: Do we need to try to parse?

File: src/service/real-time-config/real-time-config-impl.js (hunk 1 above)
Comment (calebcordry) [reply src/service/real-time-config/real-time-config-impl.js:628]: I think we can remove this, if they are using a url they should just return the json.

**Review by calebcordry — APPROVED** [review]
Thanks for taking this on!

File: src/service/real-time-config/real-time-config-impl.js [hunk 3]
```diff
@@ -502,45 +515,67 @@ export class RealTimeConfigManager {
...
-    errorReportingUrl
+    errorReportingUrl,
+    element
   ) {
+    let rtcFetch;
+    if (isAmpScriptUri(url)) {
+      rtcFetch = Services.scriptForDocOrNull(element)
+        .then((service) => {
+          userAssert(service, 'AMP-SCRIPT is not installed.');
+          return service.fetch(url);
```
Comment (calebcordry) [inline src/service/real-time-config/real-time-config-impl.js:526]: This is probably me not understanding how amp script works, but how to we ensure they are using the `no-dom` attr?

File: src/service/real-time-config/real-time-config-impl.js (hunk 2 above)
Comment (samouri) [reply src/service/real-time-config/real-time-config-impl.js:531]: Nope! I know I just named it `fetch`, but it isn't actually that much like the browser's fetch. No need to call `.text()` or `.json()` on the response. It essentially received a `postMessage`'d response, which is already JSON (or a primitive in which case we throw)

File: src/service/real-time-config/real-time-config-impl.js (hunk 3 above)
Comment (samouri) [reply src/service/real-time-config/real-time-config-impl.js:526]: From a security standpoint, I don't think we actually need it to be `nodom`. 
I.e. some pubs may want to have their amp-script perform both functions:
1. As a normal amp-script
2. As something that exports a fn for RTC

All the usual amp-script restrictions apply (re. dom mutation without user activity, etc)

Comment (samouri) [reply src/service/real-time-config/real-time-config-impl.js:524]: Going to followup on this to see if `sandboxed` should force nodom

**Pull request merged**